### PR TITLE
Add basic lemming node implementation

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: topo/node/lemming/lemming.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+func New(nodeImpl *node.Impl) (node.Node, error) {
+	if nodeImpl == nil {
+		return nil, fmt.Errorf("nodeImpl cannot be nil")
+	}
+	if nodeImpl.Proto == nil {
+		return nil, fmt.Errorf("nodeImpl.Proto cannot be nil")
+	}
+	defaults(nodeImpl.Proto)
```
Comment (alexmasi) [inline topo/node/lemming/lemming.go:34]: should use the return value here, even though it appears to modify the input in place as well

File: topo/node/lemming/lemming.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		Impl: nodeImpl,
+	}
+	return n, nil
+}
+
+type Node struct {
+	*node.Impl
+}
+
+func (n *Node) ResetCfg(ctx context.Context) error {
```
Comment (alexmasi) [inline topo/node/lemming/lemming.go:45]: PushConfig()
ResetCfg()
GenerateSelfSigned()

should all be implemented even if its just to return an "unimplemented" error with a message describing how to do the task (even if its as simple as config reset is not supported by Lemming, or config reset can be done by opening a cli on Lemming container and doing ...)

and you should include a block like: https://github.com/openconfig/kne/blob/f268e9f5c7adf28e4529dfa5aefa911124d9da23/topo/node/ceos/ceos.go#L51 to validate the signatures

File: topo/node/lemming/lemming.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		pb.Config.Command = []string{"/lemming/lemming"}
+	}
+	if len(pb.GetConfig().GetArgs()) == 0 {
+		pb.Config.Args = []string{"--alsologtostderr"}
+	}
+	if pb.Config.EntryCommand == "" {
+		pb.Config.EntryCommand = fmt.Sprintf("kubectl exec -it %s -- /bin/bash", pb.Name)
+	}
+	if pb.Services == nil {
+		pb.Services = map[uint32]*tpb.Service{
```
Comment (alexmasi) [inline topo/node/lemming/lemming.go:66]: add ssh?

File: topo/node/lemming/lemming.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+type Node struct {
+	*node.Impl
+}
+
+func (n *Node) ResetCfg(ctx context.Context) error {
+	return nil
+}
+
+func defaults(pb *tpb.Node) *tpb.Node {
```
Comment (alexmasi) [inline topo/node/lemming/lemming.go:49]: add some labels and constraints? see other node implementations

File: topo/node/lemming/lemming.go (hunk 1 above)
Comment (DanG100) [reply topo/node/lemming/lemming.go]: done

File: topo/node/lemming/lemming.go (hunk 3 above)
Comment (DanG100) [reply topo/node/lemming/lemming.go:95]: ssh isn't enabled of the lemming container, leaving it off for now

File: topo/node/lemming/lemming.go (hunk 4 above)
Comment (DanG100) [reply topo/node/lemming/lemming.go:66]: added

File: topo/node/lemming/lemming.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	_ node.ConfigPusher = (*Node)(nil)
+	_ node.Resetter     = (*Node)(nil)
+)
+
+func (n *Node) ResetCfg(ctx context.Context) error {
+	return nil
+}
+
+func (n *Node) ConfigPush(context.Context, io.Reader) error {
+	return fmt.Errorf("config push is not implemented using gNMI to configure device")
```
Comment (alexmasi) [inline topo/node/lemming/lemming.go:59]: can you make these return status.Unimplemented grpc error codes?

File: topo/node/lemming/lemming.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+// Add validations for interfaces the node provides
+var (
+	_ node.Certer       = (*Node)(nil)
+	_ node.ConfigPusher = (*Node)(nil)
+	_ node.Resetter     = (*Node)(nil)
+)
+
+func (n *Node) ResetCfg(ctx context.Context) error {
+	return nil
```
Comment (alexmasi) [inline topo/node/lemming/lemming.go:55]: maybe add a log here that resetcfg is a noop

**Review by alexmasi — APPROVED** [review]

File: topo/node/lemming/lemming.go (hunk 5 above)
Comment (DanG100) [reply topo/node/lemming/lemming.go]: done

File: topo/node/lemming/lemming.go (hunk 6 above)
Comment (DanG100) [reply topo/node/lemming/lemming.go:59]: done

**Comment by DanG100** [comment]
@marcushines What do want to do about these lint issues? it would a somewhat significant breakage to fix

**Comment by marcushines** [comment]
I would disable lint for those sections

**Comment by DanG100** [comment]
> I would disable lint for those sections and add comment to fix

Added symlink from .protolint.yaml into .github/linter to the github ci correctly used the config file. Added todo to remove the linter exclusions

File: .github/linters/.protolintrc.yml [hunk 7]
```diff
@@ -0,0 +1 @@
+../../.protolint.yaml
```
Comment (alexmasi) [inline .github/linters/.protolintrc.yml:1]: I think this file is from a time before the linter presubmit, can probably just rename it and move it here

**Review by alexmasi — APPROVED** [review]

File: .github/linters/.protolintrc.yml (hunk 7 above)
Comment (DanG100) [reply .github/linters/.protolintrc.yml]: done

**Review by alexmasi — APPROVED** [review]

**Pull request merged**